In [3]:
# import data
import pandas as pd
# CONVENTION: dataframe object name starts with df
dfjeans = pd.read_excel("../data/jeans_data.xlsx")

In [8]:
dfjeans

,week,sales,price,special_event,se_indicator
0,1,230,114.75,Presidents Day,1
1,2,71,135.00,NaN,0
2,3,70,135.00,NaN,0
3,4,65,135.00,NaN,0
4,5,60,135.00,NaN,0
5,6,195,114.75,Easter,1
6,7,76,135.00,NaN,0
7,8,51,135.00,NaN,0
8,9,79,135.00,NaN,0
9,10,146,114.75,NaN,0


In [5]:
dfjeans.describe()

,week,sales,price,se_indicator
count,26.000000,26.000000,26.000000,26.000000
mean,13.500000,92.115385,128.769231,0.153846
std,7.648529,65.149875,12.063151,0.367946
min,1.000000,46.000000,101.250000,0.000000
25%,7.250000,51.000000,135.000000,0.000000
50%,13.500000,62.500000,135.000000,0.000000
75%,19.750000,78.250000,135.000000,0.000000
max,26.000000,253.000000,135.000000,1.000000


# Model 1 Regression - Price ($) and SE_indicator

In [12]:
# Regression using Price and Special Event indicator as predictors
import statsmodels.api as sm

# Add constant to feature matrix which includes price and se_indicator
X_model1 = sm.add_constant(dfjeans[["price","se_indicator"]].astype("int"))

# Fit the OLS model
olsjeans1 = sm.OLS(dfjeans.sales, X_model1).fit()

In [13]:
# Print summary
olsjeans1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.964
Model:                            OLS   Adj. R-squared:                  0.961
Method:                 Least Squares   F-statistic:                     306.6
Date:                Tue, 24 Mar 2026   Prob (F-statistic):           2.62e-17
Time:                        12:40:19   Log-Likelihood:                -101.81
No. Observations:                  26   AIC:                             209.6
Df Residuals:                      23   BIC:                             213.4
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const          506.5693     42.491     11.922      0.000     418.671     594.468
price           -3.3113      0.320    -10.348      0.000      -3.973      -2.649
se_indicator    75.1486     10.643      7.061      0.000      53.132      97.166
==============================================================================
Omnibus:                        1.784   Durbin-Watson:                   1.241
Prob(Omnibus):                  0.410   Jarque-Bera (JB):                1.340
Skew:                           0.348   Prob(JB):                        0.512
Kurtosis:                       2.132   Cond. No.                     2.21e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.21e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## Analysis of Model 1 Regression

### Coefficients

Price returned with a coefficient of -3.31 while se_indicator had a coefficient of 75.15. We can interpret this as saying for each 1 unit increase of price ($1) we expect a decrease of about 3.31 units of sales, which lines up with common sense that as price increases, people's willingness to buy decreases (depending on the product).

On the other hand, the 75.15 coefficient on se_indicator shows that when there is a special event (se_indicator = 1) then we should expect sales to increase by 75, which shows a strong effect on the final sales number when relevant.

### P Values 
Both independent variables returned with a P-value of 0.000 (rounded to 3 digits), which indicates statistical significance. This would allow us to reject the null hypothesis that these two independent variables are not at least signficantly correleated with sales, since they are under the 0.05 threshold for 95% confidence.

### Adjusted R^2 score

For this regression, we have an adjusted R^2 score of .961, which can be interpreted as all of our predictors account for 96.1% of the variation observed in the dependent variable sales. This is quite a high score and gives at least a strong signal that this model includes features that are going to be useful for predicting sales effectively. Additionally, since it is the adjusted score, even with the penalty for including more terms (not that 2 terms is a large number by any means) it still scores highly, indicating a strong model for prediction.

# Model 2 Regression - Price as a discount and SE_indicator

In [16]:
# Regression using Price and Special Event indicator as predictors
import statsmodels.api as sm

# Create dummy variable for price as discount
dfjeans['price_discount'] = dfjeans.price / 135.00

# Add constant to feature matrix which includes price and se_indicator
X_model2 = sm.add_constant(dfjeans[["price_discount","se_indicator"]].astype("int"))

# Fit the OLS model
olsjeans2 = sm.OLS(dfjeans.sales, X_model2).fit()

In [17]:
olsjeans2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.960
Model:                            OLS   Adj. R-squared:                  0.957
Method:                 Least Squares   F-statistic:                     279.4
Date:                Tue, 24 Mar 2026   Prob (F-statistic):           7.32e-17
Time:                        12:40:53   Log-Likelihood:                -102.98
No. Observations:                  26   AIC:                             212.0
Df Residuals:                      23   BIC:                             215.7
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const            157.0000      9.549     16.441      0.000     137.246     176.754
price_discount   -98.1000     10.015     -9.795      0.000    -118.818     -77.382
se_indicator      68.7500     11.695      5.878      0.000      44.557      92.943
==============================================================================
Omnibus:                        0.114   Durbin-Watson:                   2.760
Prob(Omnibus):                  0.945   Jarque-Bera (JB):                0.030
Skew:                           0.028   Prob(JB):                        0.985
Kurtosis:                       2.843   Cond. No.                         8.36
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Analysis of Model 2 Regression

### Coefficients

In model 2, price_discount had a coefficient of -98.1 while se_indicator had 68.75. Similar to the first regression, these tell us how much change we can expect in sales when each variable moves by 1. In this case, neither of these independent variables actually move by 1 but have a maximum value of 1 and either equal 0 (se_indicator) or can range values between 0 and 1 (price_discount). 

se_indicator is the same as model 1, aside from the change in the coefficient where it is lower than the previous model, while price_discount has the effect where as the discount increases (the % of the original price goes down or in other words, price_discount goes closer to 0) the negative effect of the coefficient is reduced. It is more or less a different way of exhibiting the same behavior as the first regression; as the price goes down, sales goes up (or in this case are reduced less since it is a negative coefficient).

### P Values 
All independent variables returned with a P-value of 0.000 (rounded to 3 digits), which indicates statistical significance. This would allow us to reject the null hypothesis that these two independent variables are not at least signficantly correleated with sales, since they are under the 0.05 threshold for 95% confidence.

This result is the same as model 1, so both models have independent variables that are all statistically significant for sales, indicating no clear advantage in that regard of one model over the other.

### Adjusted R^2 score

For regression 2, we have an adjusted R^2 score of .957, which can be interpreted as all of our predictors account for 95.7% of the variation observed in the dependent variable sales. This is quite a high score and gives at least a strong signal that this model includes features that are going to be useful for predicting sales effectively.

This is just slightly lower than the r^2 term in model 1 and while it is not much worse, it at least gives us a way to potentially compare and rank our models over the other. Since we are not doing any sort of train/test splitting and predicting in this case, if r^2 is the only metric we have to use to compare the two models, there would be a slight preference for model 1, since it has better predictive power for the variance in sales over model 2 according to the slightly higher r^2 score.